packages/ui-motion/src/Transition/BaseTransition.ts

```diff
@@ -38,14 +38,37 @@ const STATES = {
   ENTERED: 2
 }
 
+type Props = {
+  in?: boolean
+  unmountOnExit?: boolean
+  transitionOnMount?: boolean
+  transitionEnter?: boolean
+  transitionExit?: boolean
+  enterDelay?: number
+  exitDelay?: number
+  transitionClassName?: string
+  exitedClassName?: string
+  exitingClassName?: string
+  enteredClassName?: string
+  enteringClassName?: string
+  onTransition?: (...args: any[]) => any
+  onEnter?: (...args: any[]) => any
+  onEntering?: (...args: any[]) => any
+  onEntered?: (...args: any[]) => any
+  onExit?: (...args: any[]) => any
+  onExiting?: (...args: any[]) => any
+  onExited?: (...args: any[]) => any
+  children?: any
+  className?: string
+}
```

Comment:
For some reason this was not typed during the TS conversion, I added here some simple typings so its more clear how this works.